pkg/controller/gardenletdeployer/actuator.go

```diff
@@ -429,64 +432,98 @@ func (a *Actuator) checkSeedSpec(ctx context.Context, spec *gardencorev1beta1.Se
 }
 
 func (a *Actuator) reconcileSeedSecrets(ctx context.Context, obj client.Object, spec *gardencorev1beta1.SeedSpec, gardenletDeployment *seedmanagementv1alpha1.GardenletDeployment) error {
-	// If backup is specified, create or update the backup secret if it doesn't exist or is owned by the object
+	if spec.Backup == nil {
+		return nil
+	}
+
+	// If backup is specified and DoNotCopyBackupCredentials feature gate is enabled,
+	// create or update the backup secret if it doesn't exist or is owned by the object.
 	// TODO(vpnachev): Add support for WorkloadIdentity
-	if spec.Backup != nil {
-		var checksum string
-
-		// Get backup secret
-		backupSecret, originalErr := kubernetesutils.GetSecretByObjectReference(ctx, a.GardenClient, spec.Backup.CredentialsRef)
-		if originalErr == nil {
-			checksum = utils.ComputeSecretChecksum(backupSecret.Data)[:8]
-		} else if client.IgnoreNotFound(originalErr) != nil {
-			return originalErr
+	var (
+		checksum string
+		copy     = !utilfeature.DefaultFeatureGate.Enabled(features.DoNotCopyBackupCredentials)
```

Comment:
This one I leave it to you, but for me it is easier to read the code later. 
```suggestion
		copyAllowed     = !utilfeature.DefaultFeatureGate.Enabled(features.DoNotCopyBackupCredentials)
```